src/appleseed-max-impl/appleseedblendmtl/appleseedblendmtl.cpp

```diff
@@ -0,0 +1,744 @@
+        m_pblock->SetValue(ParamIdBaseMtl, 0, m);
+    else
+        m_pblock->SetValue(ParamIdLayerMtl, 0, m, i-1);
+}
+
+MSTR AppleseedBlendMtl::GetSubMtlSlotName(int i)
+{
+    const auto mtl_id = static_cast<MtlId>(i);
+    return g_material_slot_names[mtl_id];
+}
+
+int AppleseedBlendMtl::get_sides() const
+{
+    return asr::ObjectInstance::BothSides;
+}
+
+bool AppleseedBlendMtl::can_emit_light() const
+{
+    return false;
+}
+
+asf::auto_release_ptr<asr::Material> AppleseedBlendMtl::create_material(
+    asr::Assembly&  assembly,
+    const char*     name,
+    const bool      use_max_procedural_maps)
+{
+    return
+        use_max_procedural_maps
+            ? create_builtin_material(assembly, name)
+            : create_osl_material(assembly, name);
+}
+
+asf::auto_release_ptr<asr::Material> AppleseedBlendMtl::create_osl_material(
+    asr::Assembly&  assembly,
+    const char*     name)
+{
+    Mtl* mat = nullptr;
+    m_pblock->GetValue(ParamIdBaseMtl, 0, mat, FOREVER);
+
+    if (!mat)
+        return asr::OSLMaterialFactory().create(name, asr::ParamArray());
+
+    auto shader_group_name = make_unique_name(assembly.shader_groups(), std::string(name) + "_shader_group");
+    auto shader_group = asr::ShaderGroupFactory::create(shader_group_name.c_str());
+
+    connect_sub_mtl(assembly, shader_group.ref(), name, "BaseMtl", mat);
+
+    TimeValue time = GetCOREInterface()->GetTime();
+    
+    int mat_count = m_pblock->Count(ParamIdLayerMtl);
+
+    asr::ParamArray shader_params;
+    int j = 0;
+    for (int i = 0; i < mat_count; ++i)
+    {
+        Mtl* mat = nullptr;
+        m_pblock->GetValue(ParamIdLayerMtl, time, mat, FOREVER, i);
+
+        if (!mat)
+            continue;
+
+        connect_sub_mtl(assembly, shader_group.ref(), name, asf::format("LayerMtl_{0}", j).c_str(), mat);
+
+        Texmap* tex = nullptr;
+        m_pblock->GetValue(ParamIdMaskTex, time, tex, FOREVER, i);
+        if (tex)
+        {
+            float mask_amount = m_pblock->GetFloat(ParamIdMaskAmount, time, FOREVER, i) / 100.0f;
+            connect_float_texture(shader_group.ref(), name, asf::format("MaskColor_{0}", j).c_str(), tex, mask_amount);
+        }
+
+        shader_params.insert(
+            asf::format("MixColor_{0}", j).c_str(), fmt_osl_expr(to_color3f(m_pblock->GetColor(ParamIdMixColor, time, FOREVER, i))));
+
+        ++j;
+    }
+    
+    // Must come last.
+    shader_group->add_shader("surface", "as_max_blend_material", name, shader_params);
+    
+    assembly.shader_groups().insert(shader_group);
+
+    //
+    // Material.
+    //
+
+    asr::ParamArray material_params;
+    material_params.insert("osl_surface", shader_group_name);
+
+    return asr::OSLMaterialFactory().create(name, material_params);
+}
+
+asf::auto_release_ptr<asr::Material> AppleseedBlendMtl::create_builtin_material(
+    asr::Assembly&  assembly,
+    const char*     name)
+{
+    return asr::GenericMaterialFactory().create(name, asr::ParamArray());
```

Comment:
There would be a way to support this material with appleseed's built-in shading system, by using a `BSDFBlend` or `BSDFMix`. We could add versions of these that take 10+1 BSDFs instead of just 2. The only thing not supported would be using different bump/normal mapping materials per layer.